### ISSUE TITLE
Unable to get OBJECTID of feature

### ISSUE THREAD

**Issue by zhangjinzhou** [naturalatlas/node-gdal#232]
I am reading data from a geodatabase. The code looks like this:
```
var dataset = gdal.open(inputPath);
var layer = dataset.layers.get(layerName);
layer.features.forEach(function (feature, featureId) {
    var geom;
    var recordId = null;
    var routeId = null;
    try {
        var geomWKB = feature.getGeometry().toWKB();
        var geomWKBString = geomWKB.toString("hex");
        geom = geomWKBString;
        // recordColumn usually has value of 'OBJECTID'
        if (recordColumn) {
            let names = feature.fields.getNames();
            recordId = feature.fields.get(recordColumn);
        }
       if (routeIdColumn) {
           routeId = feature.fields.get(routeIdColumn);
       }
    }
    catch (err) {
    }
    ......   
});
```
The code used to work. However, since this week, `feature.fields.get('OBJECTID')` is not working anymore. I listed the fields and found OBJECTID was no longer there. Then I inspected the layer itself and noticed 'OBJECTID' existed as 'FIDColumn'. How can I fix this so that every feature gets OBJECTID passed through?

**Comment by brandonreavis** [comment]
Hmm... the behavior should be consistent. Did your file change in the last week? Can you post the file before & after?

**Comment by zhangjinzhou** [comment]
@brandonreavis The file we use is exactly the same, according to our tester. How can I post a file here? It is a zipped file geodatabase. I opened it up with various GIS software and OBJECTID seems to have no issue.

**Comment by brandonreavis** [comment]
If its a small file you should be able to attach the zip to this thread, but if its large you will probably have to upload it elsewhere. Have you tried using `feature.fid` documented here: http://naturalatlas.github.io/node-gdal/classes/gdal.Feature.html#attr-fid? That doesn't explain why the behavior changed, but maybe GDAL is exposing the data that way instead of via a normal field.

**Comment by zhangjinzhou** [comment]
@brandonreavis I posted this question on stack exchange. Someone mentioned there is a current change of gdal setting the default of "preserve_fid" to "yes". This could be the reason but not 100 percent sure since I haven't updated the node-gdal library yet. It still worth a try. Is there a way to configure this option?

**Comment by zhangjinzhou** [comment]
@brandonreavis I have attached a sample data in my initial post.

**Comment by zhangjinzhou** [comment]
@brandonreavis Yeah this is the way I am using as a temporary solution. I will keep it. Thanks Brandon!
